### PR TITLE
Added all fim_db (files and registries) unit tests in the Fim registry improvement

### PR DESCRIPTION
**Pull request by jotacarma90**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/6340|

## Description

This PR adds and modifies files related to unit tests for **fim_db.c**, **fim_db_registries.c** and **fim_db_files.c**.
We created some libraries like **expect_fim_db** to auxiliary functions in the tests.

### CODE REVIEW

File: src/unit_tests/syscheckd/db/test_fim_db_files.c [hunk 1]
```diff
@@ -0,0 +1,1302 @@
...
+    expect_fim_db_bind_delete_data_id(1);
+
+    will_return(__wrap_sqlite3_step, 0);
+    will_return(__wrap_sqlite3_step, SQLITE_ERROR);
+
+    will_return(__wrap_sqlite3_errmsg,"ERROR MESSAGE");
+
+    expect_string(__wrap__merror, formatted_msg, "Step error deleting data: ERROR MESSAGE");
+
+    ret = fim_db_insert(test_data->fim_sql, test_data->entry->file_entry.path, test_data->entry->file_entry.data, test_data->saved);
```
Comment (antoniomanuelfr) [inline src/unit_tests/syscheckd/db/test_fim_db_files.c:250]: Avoid code lines longer than 120 characters.

File: src/unit_tests/syscheckd/db/test_fim_db_files.c [hunk 2]
```diff
@@ -0,0 +1,1302 @@
...
+    will_return(__wrap_sqlite3_step, SQLITE_DONE);
+#endif
+
+    // Wrap functions for fim_db_insert_data() & fim_db_insert_path()
+    expect_fim_db_insert_data_success(0);
+    expect_fim_db_insert_path_success();
+
+    expect_fim_db_check_transaction();
+
+    ret = fim_db_insert(test_data->fim_sql, test_data->entry->file_entry.path, test_data->entry->file_entry.data, test_data->saved);
```
Comment (antoniomanuelfr) [inline src/unit_tests/syscheckd/db/test_fim_db_files.c:295]: Avoid code lines longer than 120 characters.

File: src/unit_tests/syscheckd/db/test_fim_db_files.c [hunk 3]
```diff
@@ -0,0 +1,1302 @@
...
+    expect_value(__wrap_sqlite3_column_int, iCol, 0);
+    will_return(__wrap_sqlite3_column_int, 1);
+
+    // Wrap functions for fim_db_insert_data() & fim_db_insert_path()
+    expect_fim_db_insert_data_success(1);
+    expect_fim_db_insert_path_success();
+
+    expect_fim_db_check_transaction();
+
+    ret = fim_db_insert(test_data->fim_sql, test_data->entry->file_entry.path, test_data->entry->file_entry.data, test_data->saved);
```
Comment (antoniomanuelfr) [inline src/unit_tests/syscheckd/db/test_fim_db_files.c:335]: Avoid code lines longer than 120 characters.

File: src/unit_tests/syscheckd/db/test_fim_db_files.c [hunk 4]
```diff
@@ -0,0 +1,1302 @@
...
+    will_return(__wrap_sqlite3_step, SQLITE_DONE);
+
+    // Wrap functions for fim_db_insert_data() & fim_db_insert_path()
+    int inode_id = 0;
+    expect_fim_db_insert_data_success(inode_id);
+    expect_fim_db_insert_path_success();
+
+    expect_fim_db_check_transaction();
+
+    ret = fim_db_insert(test_data->fim_sql, test_data->entry->file_entry.path, test_data->entry->file_entry.data, test_data->saved);
```
Comment (antoniomanuelfr) [inline src/unit_tests/syscheckd/db/test_fim_db_files.c:380]: Avoid code lines longer than 120 characters.

File: src/unit_tests/syscheckd/db/test_fim_db_files.c [hunk 5]
```diff
@@ -0,0 +1,1302 @@
...
+    expect_fim_db_bind_path("/test/path");
+#endif
+
+    will_return(__wrap_sqlite3_step, 0);
+    will_return(__wrap_sqlite3_step, SQLITE_ERROR);
+
+    will_return(__wrap_sqlite3_errmsg, "ERROR MESSAGE");
+    expect_string(__wrap__merror, formatted_msg, "Step error getting data row: ERROR MESSAGE");
+
+    ret = fim_db_insert(test_data->fim_sql, test_data->entry->file_entry.path, test_data->entry->file_entry.data, test_data->saved);
```
Comment (antoniomanuelfr) [inline src/unit_tests/syscheckd/db/test_fim_db_files.c:404]: Avoid code lines longer than 120 characters.

**Review by antoniomanuelfr — CHANGES_REQUESTED** [review]

File: src/unit_tests/syscheckd/db/test_fim_db_files.c (hunk 1 above)
Comment (Molter73) [reply src/unit_tests/syscheckd/db/test_fim_db_files.c]: Done.

File: src/unit_tests/syscheckd/db/test_fim_db_files.c (hunk 2 above)
Comment (Molter73) [reply src/unit_tests/syscheckd/db/test_fim_db_files.c]: Done.

File: src/unit_tests/syscheckd/db/test_fim_db_files.c (hunk 3 above)
Comment (Molter73) [reply src/unit_tests/syscheckd/db/test_fim_db_files.c]: Done.

File: src/unit_tests/syscheckd/db/test_fim_db_files.c (hunk 4 above)
Comment (Molter73) [reply src/unit_tests/syscheckd/db/test_fim_db_files.c]: Done.

File: src/unit_tests/syscheckd/db/test_fim_db_files.c (hunk 5 above)
Comment (Molter73) [reply src/unit_tests/syscheckd/db/test_fim_db_files.c]: Done.

**Review by antoniomanuelfr — APPROVED** [review]
LGTM!

**Pull request merged**